src/wp-admin/includes/class-pclzip.php

```diff
@@ -4209,12 +4209,20 @@ function privExtractFileAsString(&$p_entry, &$p_string, &$p_options)
         if ($p_entry['compression'] == 0) {
 
           // ----- Reading the file
-          $p_string = @fread($this->zip_fd, $p_entry['compressed_size']);
+          if ($p_entry['compressed_size'] > 0) {
+            $p_string = @fread($this->zip_fd, $p_entry['compressed_size']);
+          } else {
+            $p_string = false;
+          }
         }
         else {
 
           // ----- Reading the file
-          $v_data = @fread($this->zip_fd, $p_entry['compressed_size']);
+          if ($p_entry['compressed_size'] > 0) {
+            $v_data = @fread($this->zip_fd, $p_entry['compressed_size']);
+          } else {
+            $v_data = false;
```

Comment:
Similar to my other comment, I'd like to suggest setting this to an empty string as well.

The below call to `gzinflate()` expects a string for the first parameter, so setting `$v_data` to `false` may cause other fatal errors in the future.

Ref: https://www.php.net/manual/en/function.gzinflate.php